### DISCUSSION TITLE
Query related to encryption key size increase

### DISCUSSION THREAD

**Discussion by dev-aarifsk** [Tencent/MMKV#1292] [Ideas]
Hello, 
* I would like to check if there are plans to allow user to choose encryption e.g. 192 bit / 256 bit. 
* Are there any known performance related concerns  on using 256 bit?
Thanks

**Comment by lingol** [comment]
Not in our plan. The main challenge is not about performance, with hardware acceleration, AES 128 has nearly the same speed as 256. 
It's the refactor / code challenge. For some optimization is tied to the size of the encryption key.

I will convert this issue to discussion as a long term feature request.

**Comment by lingol** [comment]
MMKV v2.3.0 provides AES-256 key length. http://github.com/Tencent/mmkv/releases/tag/v2.3.0
